### PR TITLE
CORE-17933 Implement fetching filtered transactions and signatures

### PR DESCRIPTION
**Pull request by jennyang-r3**
This PR contains the implementation of `findFilteredTransactionsAndSignatures` API, that fetches filtered transactions of dependencies and its notary signature and using this API in `FinalityFlow` and `ContractVerifyingNotaryClient` based on [this design](https://lucid.app/lucidchart/06502244-3d17-4b34-9e1e-10b9164eb7ff/edit?page=0_0&invitationId=inv_fe70d458-8eef-48b2-b2ac-e96b411ba8b6#)

### Test Result / Plan
**Integration Testing**
I tested the API, whether it finds the correct filtered transactions and signatures from 2 persisted transaction outputs.

**Unit test**
I tested ledger persistence service for finding filtered transactions and signatures given a notary composite key or public keys

**E2E Testing**
I tested using `VerifyingNotaryWithoutBackchain` test in E2E repo and wrote a test case where it asserts correct filtered tx ids fetched using the API
 
image

### CODE REVIEW

**Review by lankydan — DISMISSED** [review]
Browsed it and nothing jumped out to me if @nkovacsx's happy then thats ok with me.

**Review by rick-r3 — DISMISSED** [review]

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 1]
```diff
@@ -109,7 +109,7 @@ pipeline {
         }
         stage('smoketests') {
                 options {
-                    timeout(time: 30, unit: 'MINUTES')
+                    timeout(time: 100, unit: 'MINUTES')
```
Comment (dickon) [inline .ci/e2eTests/JenkinsfileCombinedWorker]: I know you are doing this for debug purposes; we shouldn't merge this change without discussion.

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 2]
```diff
@@ -109,14 +109,14 @@ pipeline {
...
                 }
                 steps {
                    gradlew('smoketest -PisCombinedWorker=true')
                 }
                 post {
                     always {
-                        junit allowEmptyResults: true, testResults: '**/test-results/**/TEST-*.xml'
+                        junit allowEmptyResults: false, testResults: '**/test-results/**/TEST-*.xml'
```
Comment (wzur-r3) [inline .ci/e2eTests/JenkinsfileCombinedWorker]: please revert that change

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt [hunk 3]
```diff
@@ -81,4 +81,16 @@ interface UtxoPersistenceService {
     )
 
     fun persistTransactionSignatures(id: String, signatures: List<ByteArray>, startingIndex: Int)
+
+    /**
+     * Retrieve filtered transactions and its signatures with the given a list of state references.
```
Comment (dickon) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt:86]: Please clarify this message; does this return multiple results in the map, and how does that relate to the `stateRefs`?

Comment (jennyang-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt:86]: yes, it returns multiple results in the map, given state refs. so the relationship of each state ref and filtered transaction, signatures will be one to one.

Comment (jennyang-r3) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt:86]: Does this make sense or more confusing?
```
     * Retrieve matching filtered transactions and signatures a list of state references.
     *
     * @param stateRefs The list of [StateRef]
     * 
     * @return A map of results of fetching filtered transaction and signatures of [StateRef]. 
     * Each [FilteredTransaction] can be null when it's not found in DB.
```

**Review by rick-r3 — DISMISSED** [review]

**Review by nkovacsx — DISMISSED** [review]

**Review by nkovacsx — APPROVED** [review]

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt (hunk 3 above)
Comment (dickon) [reply components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/UtxoPersistenceService.kt:86]: Looks good.

**Review by dickon — APPROVED** [review]

**Pull request merged**